### PR TITLE
Pc service #208

### LINKED ISSUE #208: Centralize pc service code
Pathway commons service code is duplicated across the server and client. 

- [x] Refactor codebase to use a central pathway commons service module (ideally only used by the server).
- [x] Expose the pc service code via routes in the server
- [x] Remove all other instances of pc service calls
- [x] Add functionality in CDC to call pc services (mostly for search)
- [x] Rename CDC;  I have no idea what this acronym stands for and it will be confusing to anyone who works on this codebase in the future.

### PR DESCRIPTION
**Pull request by gaelder**
- Renamed CDC to apiCaller
- Rerouted all front end call to pcServices through the backend
- Centralized all backend api calls into one module
- added api endpoints for all pc service calls

closes #208

### CODE REVIEW

File: src/server/database/update.js [hunk 1]
```diff
@@ -3,7 +3,7 @@ const uuid = require('uuid/v4');
 const hash = require('object-hash');
 const config = require('./config');
 const db = require('./utilities');
-const fetch = require('node-fetch');
+const pcServices = require('./../pcServices');
```
Comment (d2fong) [inline src/server/database/update.js]: what directory does ./../pcServices resolve to?

Comment (gaelder) [reply src/server/database/update.js]: src/server/

File: src/client/features/view/cy/events/move.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const apiCaller = require('../../../../services').apiCaller;
```
Comment (d2fong) [inline src/client/features/view/cy/events/move.js]: this file should be deleted, it now exists in common/cy/events/

File: src/client/services/apiCaller/index.js [hunk 3]
```diff
@@ -1,16 +1,43 @@
 const io = require('socket.io-client');
+const qs = require('querystring');
 let socket = io.connect('/');
 
-const CDC = {
+const apiCaller = {
   getGraphAndLayout(uri, version) {
-    return fetch(`/api/get-graph-and-layout?uri=${encodeURIComponent(uri)}&version=${version}`, {
+    return fetch(`/api/get-graph-and-layout?${qs.stringify({uri, version})}`, {
```
Comment (d2fong) [inline src/client/services/apiCaller/index.js]: does qs.stringify encode the uri?

Comment (gaelder) [reply src/client/services/apiCaller/index.js]: It creates a query string from the object and encodes all values

File: src/server/pcServices.js [hunk 4]
```diff
@@ -0,0 +1,54 @@
+let fetch = require('node-fetch');
```
Comment (d2fong) [inline src/server/pcServices.js]: Can you try to find a way to combine pcServices into server/pathway-commons?

**Pull request merged**